pipeline/test/services/service-cluster/testOpensearch.sh

```diff
@@ -0,0 +1,422 @@
+    readarray -t nodes < <(jq -c 'to_entries[]' <<<"$nodes_data")
+
+    for node in "${nodes[@]}"; do
+        node_name=$(jq '.key' <<<"$node")
+        readarray -t breakers < <(jq -c '.value.breakers | to_entries[]' <<<"$node")
+        for breaker in "${breakers[@]}"; do
+            tripped=$(jq ".value.tripped" <<<"$breaker")
+            name=$(jq ".key" <<<"$breaker")
+            if [[ $tripped == "1" ]]; then
+                no_error=false
+                debug_msg+="[DEBUG] A circuit breaker : $name has been triggered for node: $node_name\n"
+            fi
+        done
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] None of the circuit breakers has been triggered"
+    else
+        echo "failure ❌"
+        echo -e "$debug_msg"
+    fi
+
+}
+
+check_opensearch_aliases() {
+    echo -ne "Checking opensearch aliases indices mapping ... "
+    no_error=true
+    debug_msg=""
+
+    curl -sk -o /tmp/response -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/aliases"
+
+    aliases=$(awk '{print $1}' </tmp/response)
+    aliases_arr=("$aliases")
+
+    uniq_aliases=$(echo "${aliases_arr[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' ')
+
+    for alias in $uniq_aliases; do
+        if ! [[ $alias =~ ^[\.*] ]]; then
+
+            alias_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_alias/${alias}")
+            is_write_index="$(echo "$alias_data" | jq ".[].aliases.$alias.is_write_index | select(. == true)")"
+            if [[ "$is_write_index" == "" ]]; then
+                no_error=false
+                debug_msg+="[DEBUG] Alias : $alias has no write index \n"
+            fi
+
+        fi
+    done
+
+    if $no_error; then
+        echo "success ✔"
+        echo "[DEBUG] All aliases have write indices associated with them"
+    else
+        echo "failure ❌"
+        echo -e "$debug_msg"
+    fi
+
+}
+
+check_opensearch_mappings() {
+    echo -ne "Checking opensearch mappings/fields ... "
+    no_error=true
+    debug_msg="INDEX\t\t\t#FIELDS\tLIMIT\n"
+    debug_msg+="------------------------------------------------\n"
+
+    indices_data=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/_cat/indices" | awk '{print $3}' | tr '\n' ' ')
+    IFS=' ' read -ra indices <<<"$indices_data"
+    for index in "${indices[@]}"; do
+        fields_limit=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/${index}/_settings" | jq -r ".[\"${index}\"].settings.index.mapping.total_fields.limit")
+        fields_count=$(curl -sk -u admin:"${adminPassword}" -X GET "https://opensearch.${opsDomain}/${index}/_field_caps?fields=*" | jq -r ".fields | keys | length")
+
+        if [[ $fields_limit != "null" ]] && [[ $fields_count -gt $fields_limit ]]; then
+            no_error=false
+        fi
+        debug_msg+="${index}\t\t\t${fields_count}\t${fields_limit}\n"
```

Comment:
Only remark I have is if you could limit the output to only show when the current mappings reaches like 50 % of the limit just to keep focus on what's important.

Reply:
Sure, will try to do that

Reply:
Done